### PR TITLE
Autocomplete Combobox Enhancements

### PR DESCRIPTION
**Pull request by octatau**
# Critical Changes
- Utilize autocomplete combobox for object selector
- Disable field selection comboboxes in rollup editor when no options are available

# Changes
- Remove onmouseleave behavior from autocomplete combobox
- Decouple show/hide behavior in autocomplete combobox to manage visibility independently
- Add property to configure "disable" behavior on autocomplete combobox
- Add ability to specify a search threshold on the autocomplete combobox
- Display configurable message when no autocomplete options are available
- Add autocomplete combobox property to indicate the maximum number of search results that are displayed
- Add autocomplete combobox property to indicate that search is required to display options
- Update selectOptionByValue function to identify options based on exact match
- Use levenshtein distance to sort search results in autocomplete combobox

### CODE REVIEW

**Comment by aheber** [comment]
Thanks! I'll be able to take a look in the morning.

**Comment by aheber** [comment]
In `objectSelector.js` I think there are a bunch of properties that we don't use anymore. Things like iconName, selectedIconName, delayTime and a few others. If you want to clean those up now that would be great or we can merge this and then clean those up on the next round of review.

This looks great and seems to be working through my tests. Awesome work!

**Comment by aheber** [comment]
I know you have it on your todo list but if the icon loads late, it should back-fill the selection. (Object selector used to have it)

File: dlrs/main/lwc/objectSelector/objectSelector.js [hunk 1]
```diff
@@ -93,142 +94,40 @@ export default class ObjectSelector extends LightningElement {
...
-    const pillDiv = this.template.querySelector(".pillDiv");
-    if (pillDiv) {
-      pillDiv.classList.remove("slds-show");
-      pillDiv.classList.add("slds-hide");
-    }
-  }
   // method to update selected record from search result
   handleSelectedRecord(event) {
-    const selectedName = event.target.getAttribute("data-objname"); // get selected record Idrd from list
+    const selectedName = event.detail.selectedOption.value; // event.target.getAttribute("data-objname"); // get selected record Idrd from list
```
[the remark this answers is not in the document]
Comment (octatau) [reply dlrs/main/lwc/objectSelector/objectSelector.js:113]: Yes, I can include that in this in the overall cleanup of this component.

**Pull request merged**